### PR TITLE
Automate tests of Resv test suite to check the excl of node

### PR DESCRIPTION
**Pull request by shilpakodli**
#### Description
Automate tests test_multijob_on_resv_exclusive_node, test_job_with_exclusive_placement, test_job_running_on_multinode_reservation, test_job_with_exclhost_placement_inside_resv tests from Resv Test Suite.

#### Attach Test and Valgrind Logs/Output
[TestCheckNodeExclusivity_logs.txt](https://github.com/PBSPro/pbspro/files/4254895/TestCheckNodeExclusivity_logs.txt)

### CODE REVIEW

File: test/tests/functional/pbs_cray_check_node_exclusivity.py [hunk 1]
```diff
@@ -501,3 +501,134 @@ def test_reservation_request_node_ignore_excl(self):
...
+             'reserve_end': now + 40}
+        rid = self.submit_and_confirm_resv(a)
+        rid_q = rid.split('.')[0]
+        
+        self.logger.info('Waiting for reservation to start')
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, offset=10)
+        
+        a = {ATTR_q: rid_q, ATTR_l + '.select': '1:ncpus=1',
+             'Resource_List.place': 'scatter'}
```
Comment (bhroam) [inline test/tests/functional/pbs_cray_check_node_exclusivity.py]: Scattering 1 chunk isn't very interesting.  How about making the number of nodes 3 and scattering 2 chunks?

File: test/tests/functional/pbs_cray_check_node_exclusivity.py [hunk 2]
```diff
@@ -501,3 +501,134 @@ def test_reservation_request_node_ignore_excl(self):
...
+        Job inside a reservation asking for place=exclhost on host
+        will have all resources of the vnodes present on host assigned to it
+        """
+        now = int(time.time())
+        a = {'Resource_List.select': '1:ncpus=2:vntype=cray_compute',
+             'Resource_List.place': 'exclhost', 'reserve_start': now + 20,
+             'reserve_end': now + 40}
+        rid = self.submit_and_confirm_resv(a)
+        rid_q = rid.split('.')[0]
+        node = self.server.status(RESV, 'resv_nodes', id=rid)
```
Comment (bhroam) [inline test/tests/functional/pbs_cray_check_node_exclusivity.py]: You don't use node.  You are doing the status() to update the internal PTL cache.  I wouldn't even set it (here and other places).

**Comment by bhroam** [comment]
Travis is failing because some of your blink lines contain whitespace.

**Comment by shilpakodli** [comment]
Hi @bhroam Addressed your review comments. Please have a look at it. Attaching latest execution logs.
[TestCheckNodeExclusivity_logs_latest.txt](https://github.com/PBSPro/pbspro/files/4266941/TestCheckNodeExclusivity_logs_latest.txt)

**Comment by shilpakodli** [comment]
As per offline discussion with @prakashcv13, I have changed the test case in order with cray available resources. @bhroam Sorry for addressing this bit late. Please review new changes made.

**Comment by shilpakodli** [comment]
@bhroam and @prakashcv13 I have made changes to test case test_job_running_on_multinode_reservation in such a way, where reservation is submitted on two vnodes. And also submitting jobs according to available number of ncpus on vnodes.

**Comment by shilpakodli** [comment]
Attaching latest logs:
[TestCheckNodeExclusivity_latest_logs.txt](https://github.com/PBSPro/pbspro/files/4471497/TestCheckNodeExclusivity_latest_logs.txt)

File: test/tests/functional/pbs_cray_check_node_exclusivity.py [hunk 3]
```diff
@@ -563,28 +563,42 @@ def test_job_running_on_multinode_reservation(self):
...
         """
+        ncpus = []
+        vnodes = self.server.status(NODE)
+        num_vnodes = 2
+        i = 0
+        for vnode in vnodes:
+            if i < 2:
+                if vnode['resources_available.vntype'] == 'cray_compute':
+                    ncpus.append(int(vnode['resources_available.ncpus']))
+                    i += 1
```
Comment (bhroam) [inline test/tests/functional/pbs_cray_check_node_exclusivity.py:574]: Why don't you break when i == 2?  That way you don't complete the loop doing nothing.

**Comment by shilpakodli** [comment]
Attaching latest execution logs:
[TestCheckNodeExclusivity_latest_logs2.txt](https://github.com/PBSPro/pbspro/files/4477594/TestCheckNodeExclusivity_latest_logs2.txt)

**Pull request merged**